core/domain/user_services.py

```diff
@@ -656,3 +656,36 @@ def get_user_dashboard_stats(user_id):
             'total_plays': 0,
             'average_ratings': None
         }
+
+def get_weekly_dashboard_stats(user_id):
+    """Returns list which contains the dashboard stats of a user"""
+
+    model = user_models.UserStatsModel.get(user_id, strict=False)
+
+    if model and model.weekly_creator_stats_list:
+        return model.weekly_creator_stats_list
+    else:
+        return None
+
+def save_last_dashboard_stats(user_id):
+    """Save statistics for creator dashboard of a user by appending to a list
+    keyed by datetime.utcnow().date().
+
+    This method is called through a cron job which is run weekly.
+    """
+    model = user_models.UserStatsModel.get(user_id, strict=False)
+    if not model:
```

Comment:
Running the continuous job is expensive, and you can't really just "run it once" (except for in tests). Kicking off a job is basically an async call, and the caller of this function still needs a response.

Why don't you just create a new user stats model for these users?